File: algorithms/1800.maximum-ascending-subarray-sum/solution.go
func maxAscendingSum(nums []int) int {
	sums := []int{}
	tmp := []int{}

	for _, n := range nums {
		if len(tmp) > 0 {
			if tmp[len(tmp)-1] >= n {
				sums = append(sums, sum(tmp))
				tmp = []int{}
			}
		}
		tmp = append(tmp, n)
	}
	sums = append(sums, sum(tmp))

	max := 0
	for _, sum := range sums {
		if sum > max {
			max = sum
		}
	}
	return max
}

func sum(nums []int) int {
	sum := 0
	for _, n := range nums {
		sum += n
	}
	return sum
}

File: algorithms/1800.maximum-ascending-subarray-sum/solution.rs
impl Solution {
    pub fn max_ascending_sum(nums: Vec<i32>) -> i32 {
        let mut sums: Vec<i32> = vec![];
        let mut tmp: Vec<i32> = vec![];

        for n in nums {
            if tmp.len() > 0 {
                if tmp.last().unwrap() >= &n {
                    sums.push(tmp.iter().sum());
                    tmp = vec![]
                }
            }
            tmp.push(n)
        }
        sums.push(tmp.iter().sum());

        sums.into_iter().max().unwrap()
    }
}
